src/pages/o/[orgId]/surveys/[surveyId]/index.tsx

```diff
@@ -1,32 +1,346 @@
+          );
+        } else {
+          responses[questionId].options = [parseInt((form[name] as string)!)];
+        }
+      } else if (questionType == 'text') {
+        responses[questionId].response = form[name] as string;
+      }
+    }
+
+    let signature: null | {
+      email: string;
+      first_name: string;
+      last_name: string;
+    } = null;
+    // TODO: handle other signature types
+    if (form.sig == 'email') {
+      signature = {
+        email: form['sig.email'] as string,
+        first_name: form['sig.first_name'] as string,
+        last_name: form['sig.last_name'] as string,
+      };
+    }
+
+    try {
+      await apiClient.post(
+        `/api/orgs/${orgId}/surveys/${surveyId}/submissions`,
+        {
+          responses: Object.values(responses),
+          signature,
+        }
+      );
+    } catch (e) {
+      status = 'error';
+    }
+    if (status !== 'error') {
+      status = 'submitted';
+    }
+
+    if (status === 'submitted') {
+      res.writeHead(302, {
+        Location: `/o/${orgId}/surveys/${surveyId}/submitted`,
+      });
+      res.end();
+    }
+  }
 
   return {
     props: {
       orgId,
-      surveyId,
+      status,
+      survey,
     },
   };
 }, scaffoldOptions);
 
 type PageProps = {
   orgId: string;
-  surveyId: string;
+  status: FormStatus;
+  survey: ZetkinSurveyExtended;
 };
 
-const Page: FC<PageProps> = ({ orgId, surveyId }) => {
+type FormStatus = 'editing' | 'invalid' | 'error' | 'submitted';
+type SignatureOption = 'authenticated' | 'email' | 'anonymous';
+
+function RadioFormControlLabel(props: FormControlLabelProps) {
+  const radioGroup = useRadioGroup();
+
+  let checked = false;
+
+  if (radioGroup) {
+    checked = radioGroup.value === props.value;
+  }
+
+  if (checked) {
+    return (
+      <FormControlLabel
+        checked={checked}
+        sx={{
+          backgroundColor: '#fbcbd8',
+          borderRadius: '50px',
+        }}
+        {...props}
+      />
+    );
+  }
+
+  return <FormControlLabel checked={checked} {...props} />;
+}
+
+const Page: FC<PageProps> = ({ orgId, status, survey }) => {
+  const messages = useMessages(messageIds);
+
+  const [selectedOption, setSelectedOption] = useState<null | SignatureOption>(
+    null
+  );
+
+  const handleRadioChange = (value: SignatureOption) => {
+    setSelectedOption(value);
+  };
+
+  const currentUser = useCurrentUser();
+
   return (
-    <h1>
-      Page for org {orgId}, survey {surveyId}
-    </h1>
+    <Container style={{ height: '100vh' }}>
+      <Box alignItems="center" columnGap={1} display="flex" flexDirection="row">
+        <ZUIAvatar size="md" url={`/api/orgs/${orgId}/avatar`} />
+        {survey.organization.title}
+      </Box>
+
+      {status === 'error' && <ErrorMessage />}
+
+      <h1>{survey.title}</h1>
+
+      {survey.info_text && <p>{survey.info_text}</p>}
+
+      <form method="post">
+        {survey.elements.map((element) => (
+          <div key={element.id}>
```

Comment:
Minor thing, but instead of `div` we would normally use the MUI `Box` component.

Reply:
This line moved to `<SurveyElements />` in commit 7c7976e896c916a8fc7e0645ccabd69be3484388 and became a `<Box>` as per this suggestion! Link to the specific line below!

https://github.com/zetkin/app.zetkin.org/pull/1636/commits/7c7976e896c916a8fc7e0645ccabd69be3484388#diff-0db882c3159456451274878fc5ec41395843eba4035199b8a64cf58807632740R19